### PR TITLE
Loadouts Overhaul - Version 1

### PR DESCRIPTION
**Pull request by Stutternov**
## About The Pull Request

Redid faction loadouts, added some new ones to Paladin and Enclave Marine. (Specifically melee loadouts), buffed the Gunnery Sergeant's melee loadout, amended a trait on Armored Infantry and instead shifted to Gunnery sergeant as requested by Panzer, and fixed Sergeants having PA training which was an oversight by me when redoing the PA roles on Enclave.

Also misc reworks of loadouts for NCR and Legion, basically turning loadouts that you would ask yourself: "Why the fuck would I use this?" into actual competitive loadouts instead. Outlaw, town, and wasters untouched. They need their own but their loadout swaps will be easier and less controversial. 

Automatics for factions have also been reduced across the board, baring some SMGs which - while strong - are at least a tad weaker than the alternative rifles.

**Tl;dr:**
- reduces round start automatics from most factions.
- fixes unintended leftovers (i.e Enclave sergeant having PA training)
- re-adds some liked ollld loadouts from previous servers
- rebalances some loadouts where it was like: "why tf would you choose this one?"
- reduces amount of automatics factions get on round-start to prevent walls of bullets without having to do anything to get said automatics
- 'Removed' Senior Paladin, added Head Paladin. Basically just a role-swap, reason being head paladins have actual command authority over warrior caste instead of head scribe only. (Senior paladin pins moved to regular paladin, incase people wanna rp being 'seniors')

## Why It's Good For The Game

We keep doing changes to guns and certain "people" keep giving their favorite factions shit while neglecting entire other factions to the point it becomes onesided crap.

This only half fixes it because I really cannot be fucked to deal with players whining and moaning if I dare to even do as another coder or player suggests because giving feedback such as: "I think x is better for y reason" you get: "What asshole retard coder did this?" or "TRYING TO KILL MY FAVORITE FACTION!!!!!"

As such it's not fucking worth fixing the slop since players prefer to wallow in it.

## Pre-Merge Checklist

- [ ] You tested this on a local server.
- [ ] This code did not runtime during testing.
- [ ] You documented all of your changes.

## Changelog

:cl:
add: New Enclave Marine and Paladin melee loadouts
add: Adds big leagues book to Head Paladin melee and Gunnery Sergeant melee loadout options.
add: New senior knight scout armor for Brotherhood of Steel.
tweak: Replaces Senior Paladin with Head Paladin. Overall, basically just a name-swap with no real other changes. This is to avoid stupid caste conflict crap.
balance: Redid Legion, NCR, and some misc loadouts.
tweak: Moved trekking from armored infantry over to gunnery sergeant.
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/legion.dm [hunk 1]
```diff
@@ -428,17 +416,15 @@ Discuss balance and documentation changes with Dragonfruits#1913 or forward them
...
 	name = "Mark of the Brave"
 	head = /obj/item/clothing/head/helmet/f13/legion/vet/decan
-	suit_store = /obj/item/gun/ballistic/shotgun/automatic/combat/shotgunlever
+	suit_store = /obj/item/gun/ballistic/automatic/shotgun/riot
 	backpack_contents = list(
-		/obj/item/gun/ballistic/revolver/ballisticfist = 1,
-		/obj/item/ammo_box/shotgun/buck = 2,
-		/obj/item/gun/ballistic/revolver/m29 = 1,
+		/obj/item/melee/onehanded/machete/spatha = 1,
+		/obj/item/ammo_box/magazine/d12g = 3,
```
Comment (mazzinsanity) [inline code/modules/jobs/job_types/legion.dm:427]: Just give 2 magazines for consistency since a whole bunch of other roles get 2. Or give 3 magazines to every leadership role in the game.

Comment (Stutternov) [reply code/modules/jobs/job_types/legion.dm]: Wasn't the case for majority of NCR roles, did it anyway. There.

**Comment by Stutternov** [comment]
Undid part of the changes as requested for Legion, reduced NCR mags down from 3 to 2 as Mazz requested, re-added Vex slot because I cannot be fucked to try and appease players. They can keep their slop as is instead of bothering to make something interesting to address half their complaints.

**Review by maaacha — APPROVED** [review]
